mountpoint-s3-client/src/s3_crt_client/put_object.rs

```diff
@@ -233,7 +226,7 @@ pub struct S3PutObjectRequest {
     start_time: Instant,
     total_bytes: u64,
     /// Headers of the CompleteMultipartUpload response, available after the request was finished
-    response_headers: Arc<Mutex<Option<Headers>>>,
+    response_headers: Receiver<Headers>,
```

Comment:
I would prefer `response_headers_receiver` and update the comment.

Reply:
Pushing back a little on this. A `Receiver` is a future and that's how it is used here. But happy to clarify in the comment.

Reply:
It's just not clear from the naming that we have to call `.await` to really get the headers, but not blocking on it.